hazelcast/src/test/java/com/hazelcast/client/AuthenticationInformationLeakTest.java

```diff
@@ -167,11 +169,17 @@ private ClientMessage getKeyValue(SerializationService ss, OutputStream os, Inpu
         ClientMessage msg = MapGetCodec.encodeRequest("mapName", keyData, 0);
         msg.setPartitionId(getPartitionId(keyData));
         ClientTestUtil.writeClientMessage(os, msg);
-        ClientMessage res = ClientTestUtil.readResponse(is);
-        if (res.getMessageType() != ErrorsCodec.EXCEPTION_MESSAGE_TYPE) {
-            assertEquals(res.getMessageType(), MapGetCodec.RESPONSE_MESSAGE_TYPE);
+        AtomicReference<ClientMessage> res = new AtomicReference<>();
+        // The server sends an exception message and immediately closes the connection without waiting the message to
+        // be written. It's possible that we receive EOF from socket, which will make ClientTestUtil.readResponse throw.
+        // Therefore, we use assertTrueEventually here.
+        assertTrueEventually(() -> {
+            res.set(ClientTestUtil.readResponse(is));
+        });
```

Comment:
How is this going to work when the server closes the socket? You will not be able to read the message, and I assume `ClientTestUtil.readResponse` will throw an `EOFException`.

Reply:
Right, we should put a assertTrueEventually to the general request-response logic.

Reply:
Moved assertTrueEventually to the test logic in a way that it covers socket creation, doing the request, getting the response and asserting.